### DISCUSSION TITLE
api via https

### DISCUSSION THREAD

**Discussion by obecny** [bluenviron/mediamtx#3146] [Questions]
### Question

Does the mediamtx supports api via https instead of http ? if so can you add example, if not can you please share if such feature is already on your roadmap and if not what should happen to add such feature ?

thx a lot

**Comment by alexandrecastro-bib** [comment]
Hello @obecny ,

I have the same question and nobody answered as you can see : [my question](https://github.com/bluenviron/mediamtx/discussions/3156)

It would be sad that the api can not be on HTTPS as all streams managed secured communication.
I search a lot on the documentation and mediamtx options in config file and did not find anything. Maybe i missed something but i would like to have an answer @aler9. 

Regards and thanks in advance

**Reply by aler9** [reply]
Hello, HTTPS support in API and playback has been delayed until the implementation of an authentication system for both, and that was added in v1.6.0, so there's no reason not to add support for HTTPS, it's just the fact that now I have no time for it.
Implementing the feature is straightforward so I'm willing to review and accept any patch that provides it.

**Reply by alexandrecastro-bib** [reply]
Thanks for reply @aler9

**Reply by alexandrecastro-bib** [reply]
I'm not a go developper but i will try to add this feature if i can have any help on where is located the file to edit.

[a comment by an automated account is not shown]
